### PR TITLE
[Doc] fixing ray.remote doc to hider type-check helper overloads

### LINKED ISSUE #25641: [Core] ray.remote documentation improvement
### Description

Screen Shot 2022-06-09 at 5 59 46 PM

Currently, ray.remote is flooded by this type of annotation. We need to simplify it. 

Also, I realized some the arguments are not documented (not sure if it is intended). For example, `max_pending_calls` or `get_if_exists`. It might be just because it is not a public API though.

### Link

_No response_

### PR DESCRIPTION
**Pull request by scv119**
## Why are these changes needed?

Currently the ray.remote doc shows helper functions that only used for type-checking. Move them to a separate file and exclude from docs.
before
https://docs.ray.io/en/latest/ray-core/api/doc/ray.remote.html#ray.remote
after
https://anyscale-ray--33243.com.readthedocs.build/en/33243/ray-core/api/doc/ray.remote.html

image

## Related issue number
Closes #25641

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -157,6 +157,10 @@
     "_build",
     "source/workflows/api/doc/ray.workflow.*",
     "source/serve/api/doc/ray.serve.*",
+    # exclude ray.remote type checking overloads from docs.
+    # refer to https://github.com/ray-project/ray/pull/33243
+    # for more context.
+    "../python/ray/_private/worker.pyi",
```
Comment (jjyao) [inline doc/source/conf.py:163]: why `pyi` instead of normal `py` file?

**Review by jjyao — COMMENTED** [review]
Oh nice. I tried to fix it but couldn't figure out a way.

I assumed that you checked the type-check and it still works?

File: doc/source/conf.py (hunk 1 above)
Comment (scv119) [reply doc/source/conf.py:163]: looks that's the convention of writing interfaces.. and it was worker.pyi and removed by https://github.com/ray-project/ray/pull/25439

**Comment by scv119** [comment]
> I assumed that you checked the type-check and it still works?
I did run `/scripts/format.sh` and it doesn't complain, is there anything else i should check?

**Comment by jjyao** [comment]
If you look at the original PR that introduced the code, there are some tests we can do: basically make sure tool complains when user uses ray.remote wrong.

**Comment by scv119** [comment]
seems with worker.pyi the type in worker.py no longer works... this is a bummer.

**Pull request closed without merging**